From 089912b1c8f1dab156a699e70e461076df28de97 Mon Sep 17 00:00:00 2001
From: Jerry Snitselaar <jsnitsel@redhat.com>
Date: Sat, 13 Apr 2024 00:02:28 -0700
Subject: [PATCH 2/3] accel-config/test: Make verbose logging optional
'Content-type: text/plain'

Add option to test scripts to turn on verbose logging using
the option --verbose, and default to bash shopt +x, and
not passing -v to the test executable.

Signed-off-by: Jerry Snitselaar <jsnitsel@redhat.com>
---
 test/dsa_config_test_runner.sh |  41 +++++-----
 test/dsa_user_test_runner.sh   |  53 +++++++------
 test/iaa_user_test_runner.sh   | 134 +++++++++++++++++----------------
 3 files changed, 125 insertions(+), 103 deletions(-)

diff --git a/test/dsa_config_test_runner.sh b/test/dsa_config_test_runner.sh
index aa02864a3d74..740648a91d94 100755
--- a/test/dsa_config_test_runner.sh
+++ b/test/dsa_config_test_runner.sh
@@ -1,9 +1,16 @@
-#!/bin/bash -Ex
+#!/bin/bash -E
 # SPDX-License-Identifier: GPL-2.0
 # Copyright(c) 2019-2020 Intel Corporation. All rights reserved.
 
 . "$(dirname "$0")/common"
 
+if [[ $* =~ "--verbose" ]]; then
+    set -x
+    VERBOSE="-v"
+else
+    VERBOSE=""
+fi
+
 DSA=dsa0
 WQ0=wq0.0
 GRP0=group0.0
@@ -178,8 +185,8 @@ wq_config_test()
 		fi
 		"$ACCFG" enable-device $DSA
 		"$ACCFG" enable-wq $DSA/$WQ0
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x3 -v && echo "shoudl fail, but pass" && exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x4 -c 2 &&
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x3 "${VERBOSE}" && echo "shoudl fail, but pass" && exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x4 -c 2 "${VERBOSE}" &&
 			echo "shoudl fail, but pass" && exit 1
 		"$ACCFG" disable-device $DSA
 
@@ -191,9 +198,9 @@ wq_config_test()
 		fi
 		"$ACCFG" enable-device $DSA
 		"$ACCFG" enable-wq $DSA/$WQ0
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x3 || echo "shoudl pass, but fail" || exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x5 && echo "shoudl fail, but pass" && exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x3 -c 2 &&
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x3 "${VERBOSE}" || echo "shoudl pass, but fail" || exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x5 "${VERBOSE}" && echo "shoudl fail, but pass" && exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x3 -c 2 "${VERBOSE}" &&
 			echo "shoudl fail, but pass" && exit 1
 		"$ACCFG" disable-device $DSA
 
@@ -205,18 +212,18 @@ wq_config_test()
 		fi
 		"$ACCFG" enable-device $DSA
 		"$ACCFG" enable-wq $DSA/$WQ0
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x0 && echo "shoudl fail, but pass" && exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x2 && echo "shoudl fail, but pass" && exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x3 && echo "shoudl fail, but pass" && exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x7 && echo "shoudl fail, but pass" && exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x8 && echo "shoudl fail, but pass" && exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x4 || echo "shoudl pass, but fail" || exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x5 || echo "shoudl pass, but fail" || exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x6 || echo "shoudl pass, but fail" || exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x9 || echo "shoudl fail, but pass" || exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x5 -c 2 ||
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x0 "${VERBOSE}" && echo "shoudl fail, but pass" && exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x2 "${VERBOSE}" && echo "shoudl fail, but pass" && exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x3 "${VERBOSE}" && echo "shoudl fail, but pass" && exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x7 "${VERBOSE}" && echo "shoudl fail, but pass" && exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x8 "${VERBOSE}" && echo "shoudl fail, but pass" && exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x4 "${VERBOSE}" || echo "shoudl pass, but fail" || exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x5 "${VERBOSE}" || echo "shoudl pass, but fail" || exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x6 "${VERBOSE}" || echo "shoudl pass, but fail" || exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x9 "${VERBOSE}" || echo "shoudl fail, but pass" || exit 1
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x5 -c 2 "${VERBOSE}" ||
 			echo "shoudl pass, but fail" || exit 1
-		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x9 -c 2 ||
+		"$DSATEST" -w 0 -l 4096 -f 0x1 -o 0x1 -b 0x9 -c 2 "${VERBOSE}" ||
 			echo "shoudl pass, but fail" || exit 1
 		"$ACCFG" disable-device $DSA
 	fi
diff --git a/test/dsa_user_test_runner.sh b/test/dsa_user_test_runner.sh
index 1e6d3a602a49..2848feda68bb 100755
--- a/test/dsa_user_test_runner.sh
+++ b/test/dsa_user_test_runner.sh
@@ -1,4 +1,4 @@
-#!/bin/bash -Ex
+#!/bin/bash -E
 # SPDX-License-Identifier: GPL-2.0
 # Copyright(c) 2019-2020 Intel Corporation. All rights reserved.
 
@@ -6,15 +6,22 @@
 
 rc="$EXIT_SKIP"
 
-input1=$1
-if [ "$input1" == "--skip-config" ]; then
-DEV=`ls /dev/dsa/ | sed -ne 's|wq\([^.]\+\)\(.*\)|dsa\1/wq\1\2|p'`
-DSA=`echo $DEV | cut -f1 -d/`
-echo "$DEV"
-echo "$DSA"
+if [[ $* =~ "--verbose" ]]; then
+    VERBOSE="-v"
+    set -x
 else
-DSA=dsa0
-echo "$DSA"
+    VERBOSE=""
+fi
+
+if [[ $* =~ "--skip-config" ]]; then
+    SKIPCONFIG="true"
+    DEV=`ls /dev/dsa/ | sed -ne 's|wq\([^.]\+\)\(.*\)|dsa\1/wq\1\2|p'`
+    DSA=`echo $DEV | cut -f1 -d/`
+    echo "$DEV"
+    echo "$DSA"
+else
+    DSA=dsa0
+    echo "$DSA"
 fi
 WQ0=wq0.0
 WQ1=wq0.1
@@ -78,12 +85,12 @@ test_op()
 		echo "Performing $wq_mode_name WQ $op_name testing"
 		for xfer_size in $SIZE_1 $SIZE_4K $SIZE_64K $SIZE_1M $SIZE_2M; do
 			echo "Testing $xfer_size bytes"
-			if [ "$input1" == "--skip-config" ]; then
+			if ! test -z "${SKIPCONFIG}"; then
 			"$DSATEST" -l "$xfer_size" -o "$opcode" \
-				-f "$flag" t200 -v -d "$DEV"
+				-f "$flag" t200 "${VERBOSE}" -d "$DEV"
 			else
 			"$DSATEST" -w "$wq_mode_code" -l "$xfer_size" -o "$opcode" \
-				-f "$flag" t200 -v
+				-f "$flag" t200 "${VERBOSE}"
 			fi
 		done
 	done
@@ -112,12 +119,12 @@ test_op_batch()
 		echo "Performing $wq_mode_name WQ batched $op_name testing"
 		for xfer_size in $SIZE_1 $SIZE_4K $SIZE_64K $SIZE_1M $SIZE_2M; do
 			echo "Testing $xfer_size bytes"
-			if [ "$input1" == "--skip-config" ]; then
+			if ! test -z "${SKIPCONFIG}"; then
 			"$DSATEST" -l "$xfer_size" -o 0x1 -b "$opcode" \
-				-c 16 -f "$flag" t2000 -v -d "$DEV"
+				-c 16 -f "$flag" t2000 "${VERBOSE}" -d "$DEV"
 			else
 			"$DSATEST" -w "$wq_mode_code" -l "$xfer_size" -o 0x1 -b "$opcode" \
-				-c 16 -f "$flag" t2000 -v
+				-c 16 -f "$flag" t2000 "${VERBOSE}"
 			fi
 		done
 	done
@@ -142,12 +149,12 @@ test_dif_op()
 		echo "Performing $wq_mode_name WQ $op_name testing"
 		for xfer_size in $SIZE_512 $SIZE_1K $SIZE_4K; do
 			echo "Testing $xfer_size bytes"
-			if [ "$input1" == "--skip-config" ]; then
+			if ! test -z "${SKIPCONFIG}"; then
 			"$DSATEST" -l "$xfer_size" -o "$opcode" \
-				-f "$flag" t200 -v -d "$DEV"
+				-f "$flag" t200 "${VERBOSE}" -d "$DEV"
 			else
 			"$DSATEST" -w "$wq_mode_code" -l "$xfer_size" -o "$opcode" \
-				-f "$flag" t200 -v
+				-f "$flag" t200 "${VERBOSE}"
 			fi
 		done
 	done
@@ -172,17 +179,17 @@ test_dif_op_batch()
 		echo "Performing $wq_mode_name WQ batched $op_name testing"
 		for xfer_size in $SIZE_512 $SIZE_1K $SIZE_4K; do
 			echo "Testing $xfer_size bytes"
-			if [ "$input1" == "--skip-config" ]; then
+			if ! test -z "${SKIPCONFIG}"; then
 			"$DSATEST" -l "$xfer_size" -o 0x1 -b "$opcode" \
-				-c 16 -f "$flag" t2000 -v -d "$DEV"
+				-c 16 -f "$flag" t2000 "${VERBOSE}" -d "$DEV"
 			else
 			"$DSATEST" -w "$wq_mode_code" -l "$xfer_size" -o 0x1 -b "$opcode" \
-				-c 16 -f "$flag" t2000 -v
+				-c 16 -f "$flag" t2000 "${VERBOSE}"
 			fi
 		done
 	done
 }
-if [ "$input1" != "--skip-config" ]; then
+if test -z "${SKIPCONFIG}"; then
 _cleanup
 start_dsa
 enable_wqs
@@ -219,7 +226,7 @@ for opcode in "0x12" "0x13" "0x14" "0x15"; do
 	test_dif_op_batch $opcode $flag
 done
 
-if [ "$input1" != "--skip-config" ]; then
+if ! $SKIPCONFIG; then
 disable_wqs
 stop_dsa
 _cleanup
diff --git a/test/iaa_user_test_runner.sh b/test/iaa_user_test_runner.sh
index eb9629b4341e..f978b68505e8 100755
--- a/test/iaa_user_test_runner.sh
+++ b/test/iaa_user_test_runner.sh
@@ -1,4 +1,4 @@
-#!/bin/bash -Ex
+#!/bin/bash -E
 # SPDX-License-Identifier: GPL-2.0
 # Copyright(c) 2019-2020 Intel Corporation. All rights reserved.
 
@@ -7,13 +7,21 @@
 rc="$EXIT_SKIP"
 
 DEV_OPT=""
-input1=$1
-if [ "$input1" == "--skip-config" ]; then
+
+if [[ $* =~ "--verbose" ]]; then
+	VERBOSE="-v"
+	set -x
+else
+	VERBOSE=""
+fi
+
+if [[ $* =~ "--skip-config" ]]; then
 	DEV=`ls /dev/iax/ | sed -ne 's|wq\([^.]\+\)\(.*\)|iax\1/wq\1\2|p'`
 	DEV=`echo $DEV | cut -f1 -d' '`
 	IAA=`echo $DEV | cut -f1 -d/`
 	DEV_OPT="-d $DEV"
 	echo "$DEV"
+        SKIPCONFIG="true"
 else
 	IAA=iax1
 fi
@@ -140,10 +148,10 @@ test_op()
 			if [ "$extra_flag" != "" ]
 			then
 				"$IAATEST" -w "$wq_mode_code" -l "$xfer_size" -o "$opcode" \
-					-f "$flag" -1 "$extra_flag" -t 5000 -v $DEV_OPT
+					-f "$flag" -1 "$extra_flag" -t 5000 "${VERBOSE}" $DEV_OPT
 			else
 				"$IAATEST" -w "$wq_mode_code" -l "$xfer_size" -o "$opcode" \
-					-f "$flag" -t 5000 -v $DEV_OPT
+					-f "$flag" -t 5000 "${VERBOSE}" $DEV_OPT
 			fi
 		done
 	done
@@ -160,133 +168,133 @@ test_op_filter()
 
 		if [ $((IAA_OPCODE_MASK_SCAN & OP_CAP2)) -ne 0 ]; then
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x7c \
-				-3 128 -o $IAA_OPCODE_SCAN -t 5000 -v $DEV_OPT
+				-3 128 -o $IAA_OPCODE_SCAN -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x7c \
-				-3 256 -o $IAA_OPCODE_SCAN -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_SCAN -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 4096 -2 0x7c \
-				-3 1024 -o $IAA_OPCODE_SCAN -t 5000 -v $DEV_OPT
+				-3 1024 -o $IAA_OPCODE_SCAN -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 65536 -2 0x7c \
-				-3 16384 -o $IAA_OPCODE_SCAN -t 5000 -v $DEV_OPT
+				-3 16384 -o $IAA_OPCODE_SCAN -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1048576 -2 0x7c \
-				-3 262144 -o $IAA_OPCODE_SCAN -t 5000 -v $DEV_OPT
+				-3 262144 -o $IAA_OPCODE_SCAN -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 2097152 -2 0x7c \
-				-3 524288 -o $IAA_OPCODE_SCAN -t 5000 -v $DEV_OPT
+				-3 524288 -o $IAA_OPCODE_SCAN -t 5000 "${VERBOSE}" $DEV_OPT
 		fi
 
 		if [ $((IAA_OPCODE_MASK_SET_MEMBERSHIP & OP_CAP2)) -ne 0 ]; then
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x38 \
-				-3 256 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x38 \
-				-3 512 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 -v $DEV_OPT
+				-3 512 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 4096 -2 0x38 \
-				-3 2048 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 -v $DEV_OPT
+				-3 2048 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 65536 -2 0x38 \
-				-3 32768 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 -v $DEV_OPT
+				-3 32768 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1048576 -2 0x38 \
-				-3 524288 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 -v $DEV_OPT
+				-3 524288 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 2097152 -2 0x38 \
-				-3 1048576 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 -v $DEV_OPT
+				-3 1048576 -o $IAA_OPCODE_SET_MEMBERSHIP -t 5000 "${VERBOSE}" $DEV_OPT
 		fi
 
 		if [ $((IAA_OPCODE_MASK_EXTRACT & OP_CAP2)) -ne 0 ]; then
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x7c \
-				-3 128 -o $IAA_OPCODE_EXTRACT -t 5000 -v $DEV_OPT
+				-3 128 -o $IAA_OPCODE_EXTRACT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x7c \
-				-3 256 -o $IAA_OPCODE_EXTRACT -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_EXTRACT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 4096 -2 0x7c \
-				-3 1024 -o $IAA_OPCODE_EXTRACT -t 5000 -v $DEV_OPT
+				-3 1024 -o $IAA_OPCODE_EXTRACT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 65536 -2 0x7c \
-				-3 16384 -o $IAA_OPCODE_EXTRACT -t 5000 -v $DEV_OPT
+				-3 16384 -o $IAA_OPCODE_EXTRACT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1048576 -2 0x7c \
-				-3 262144 -o $IAA_OPCODE_EXTRACT -t 5000 -v $DEV_OPT
+				-3 262144 -o $IAA_OPCODE_EXTRACT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 2097152 -2 0x7c \
-				-3 524288 -o $IAA_OPCODE_EXTRACT -t 5000 -v $DEV_OPT
+				-3 524288 -o $IAA_OPCODE_EXTRACT -t 5000 "${VERBOSE}" $DEV_OPT
 		fi
 
 		if [ $((IAA_OPCODE_MASK_SELECT & OP_CAP2)) -ne 0 ]; then
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x7c \
-				-3 128 -o $IAA_OPCODE_SELECT -t 5000 -v $DEV_OPT
+				-3 128 -o $IAA_OPCODE_SELECT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x7c \
-				-3 256 -o $IAA_OPCODE_SELECT -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_SELECT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 4096 -2 0x7c \
-				-3 1024 -o $IAA_OPCODE_SELECT -t 5000 -v $DEV_OPT
+				-3 1024 -o $IAA_OPCODE_SELECT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 65536 -2 0x7c \
-				-3 16384 -o $IAA_OPCODE_SELECT -t 5000 -v $DEV_OPT
+				-3 16384 -o $IAA_OPCODE_SELECT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1048576 -2 0x7c \
-				-3 262144 -o $IAA_OPCODE_SELECT -t 5000 -v $DEV_OPT
+				-3 262144 -o $IAA_OPCODE_SELECT -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 2097152 -2 0x7c \
-				-3 524288 -o $IAA_OPCODE_SELECT -t 5000 -v $DEV_OPT
+				-3 524288 -o $IAA_OPCODE_SELECT -t 5000 "${VERBOSE}" $DEV_OPT
 		fi
 
 		if [ $((IAA_OPCODE_MASK_RLE_BURST & OP_CAP2)) -ne 0 ]; then
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x1c \
-				-3 512 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 512 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x1c \
-				-3 1024 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 1024 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 4096 -2 0x1c \
-				-3 4096 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 4096 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 32768 -2 0x1c \
-				-3 32768 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 32768 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 65536 -2 0x1c \
-				-3 65536 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 65536 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 131072 -2 0x1c \
-				-3 131072 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 131072 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 32 -2 0x3c \
-				-3 16 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 16 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 64 -2 0x3c \
-				-3 32 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 32 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 128 -2 0x3c \
-				-3 64 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 64 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 256 -2 0x3c \
-				-3 128 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 128 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x3c \
-				-3 256 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x3c \
-				-3 512 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 512 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 64 -2 0x7c \
-				-3 16 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 16 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 128 -2 0x7c \
-				-3 32 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 32 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 256 -2 0x7c \
-				-3 64 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 64 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x7c \
-				-3 128 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 128 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x7c \
-				-3 256 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 2048 -2 0x7c \
-				-3 512 -o $IAA_OPCODE_RLE_BURST -t 5000 -v $DEV_OPT
+				-3 512 -o $IAA_OPCODE_RLE_BURST -t 5000 "${VERBOSE}" $DEV_OPT
 		fi
 
 		if [ $((IAA_OPCODE_MASK_FIND_UNIQUE & OP_CAP2)) -ne 0 ]; then
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x38 \
-				-3 256 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x38 \
-				-3 512 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 -v $DEV_OPT
+				-3 512 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 4096 -2 0x38 \
-				-3 2048 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 -v $DEV_OPT
+				-3 2048 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 65536 -2 0x38 \
-				-3 32768 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 -v $DEV_OPT
+				-3 32768 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1048576 -2 0x38 \
-				-3 524288 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 -v $DEV_OPT
+				-3 524288 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 2097152 -2 0x38 \
-				-3 1048576 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 -v $DEV_OPT
+				-3 1048576 -o $IAA_OPCODE_FIND_UNIQUE -t 5000 "${VERBOSE}" $DEV_OPT
 		fi
 
 		if [ $((IAA_OPCODE_MASK_EXPAND & OP_CAP2)) -ne 0 ]; then
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 512 -2 0x7c \
-				-3 128 -o $IAA_OPCODE_EXPAND -t 5000 -v $DEV_OPT
+				-3 128 -o $IAA_OPCODE_EXPAND -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1024 -2 0x7c \
-				-3 256 -o $IAA_OPCODE_EXPAND -t 5000 -v $DEV_OPT
+				-3 256 -o $IAA_OPCODE_EXPAND -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 4096 -2 0x7c \
-				-3 1024 -o $IAA_OPCODE_EXPAND -t 5000 -v $DEV_OPT
+				-3 1024 -o $IAA_OPCODE_EXPAND -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 65536 -2 0x7c \
-				-3 16384 -o $IAA_OPCODE_EXPAND -t 5000 -v $DEV_OPT
+				-3 16384 -o $IAA_OPCODE_EXPAND -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 1048576 -2 0x7c \
-				-3 262144 -o $IAA_OPCODE_EXPAND -t 5000 -v $DEV_OPT
+				-3 262144 -o $IAA_OPCODE_EXPAND -t 5000 "${VERBOSE}" $DEV_OPT
 			"$IAATEST" -w "$wq_mode_code" -f "$flag" -l 2097152 -2 0x7c \
-				-3 524288 -o $IAA_OPCODE_EXPAND -t 5000 -v $DEV_OPT
+				-3 524288 -o $IAA_OPCODE_EXPAND -t 5000 "${VERBOSE}" $DEV_OPT
 		fi
 	done
 }
@@ -308,7 +316,7 @@ test_op_crypto()
 			echo "Testing $xfer_size bytes"
 
 			"$IAATEST" -w "$wq_mode_code" -l "$xfer_size" -o "$opcode" \
-				-f "$flag" -a "$aecs_flag" -t 5000 -v $DEV_OPT
+				-f "$flag" -a "$aecs_flag" -t 5000 "${VERBOSE}" $DEV_OPT
 		done
 	done
 }
@@ -329,12 +337,12 @@ test_op_transl_fetch()
 			echo "Testing $xfer_size bytes"
 
 			"$IAATEST" -w "$wq_mode_code" -l "$xfer_size" -o "$opcode" \
-				-f "$flag" -t 5000 -v $DEV_OPT
+				-f "$flag" -t 5000 "${VERBOSE}" $DEV_OPT
 		done
 	done
 }
 
-if [ "$input1" != "--skip-config" ]; then
+if test -z "${SKIPCONFIG}"; then
 	_cleanup
 	start_iaa
 	enable_wqs
@@ -487,7 +495,7 @@ if [ $((IAA_OPCODE_MASK_TRANSL_FETCH & OP_CAP0)) -ne 0 ]; then
 	test_op_transl_fetch $IAA_OPCODE_TRANSL_FETCH $flag
 fi
 
-if [ "$input1" != "--skip-config" ]; then
+if test -z "${SKIPCONFIG}"; then
 	disable_wqs
 	stop_iaa
 	_cleanup
-- 
2.44.0

